improve WORKSPACE

BowNV opened the pull request:
During testing, the following issue was reported
"
[235 / 251] Compiling external/com_google_protobuf/src/google/protobuf/descriptor.pb.cc [for host]; 2s linux-sandbox ... (7 actions running)
[289 / 298] Compiling cudnn_conv_test.cc; 0s linux-sandbox ... (7 actions running)
ERROR: .../cuda/_tests/google_cudnn_test/codes/nvidia_libs_test-master/BUILD:110:1: C++ compilation of rule '//:cudnn_test' failed (Exit 1) gcc failed: error executing command /usr/bin/gcc -U_FORTIFY_SOURCE -fstack-protector -Wall -Wunused-but-set-parameter -Wno-free-nonheap-object -fno-omit-frame-pointer '-std=c++0x' -MD -MF ... (remaining 103 argument(s) skipped)
Use --sandbox_debug to see verbose messages from the sandbox
cudnn_conv_test.cc:1203:25: error: expected constructor, destructor, or type conversion before '(' token
INSTANTIATE_TEST_SUITE_P(
^
cudnn_conv_test.cc:1208:25: error: expected constructor, destructor, or type conversion before '(' token
INSTANTIATE_TEST_SUITE_P(Conv2dFilter3x3, ConvolutionTest,
^
cudnn_conv_test.cc:1242:1: error: expected '}' at end of input
} // namespace nvidia_libs_test
^
cudnn_conv_test.cc:1242:1: error: expected '}' at end of input
cudnn_conv_test.cc: In function 'double nvidia_libs_test::{anonymous}::GetToleranceScale(const nvidia_libs_test::proto::ConvolutionConfig&)':
cudnn_conv_test.cc:148:1: warning: control reaches end of non-void function [-Wreturn-type]
}
^
"

Since INSTANTIATE_TEST_SUITE_P can be supported by master branch, I created this request for approval.  And Sha256 was generated based on today's googletest master and local testing was done and passed! 

Thanks
Bo

gawain102000 committed:
improve WORKSPACE

chsigg commented on WORKSPACE:
```diff
@@ -11,9 +11,9 @@ http_archive(
-    url = "https://github.com/google/googletest/archive/release-1.8.0.zip",
-    sha256 = "f3ed3b58511efd272eb074a3a6d6fb79d7c2e6a0e374323d1e6bcbcc1ef141bf",
-    strip_prefix = "googletest-release-1.8.0",
+    url = "https://github.com/google/googletest/archive/master.zip",
```
It should say something like 'https://github.com/google/googletest/archive/88f0493098c8d9fd0f096c2158a0e56deb952d53.zip'

'master' is just the latest.

BowNV replied:
OK, Thanks a lot! I will submit a new.

But by what command can I know that the latest master zip is 88f0493098c8d9fd0f096c2158a0e56deb952d53.zip? hope to get your guidance. 

Thanks
Bo

jlebar replied:
Use the command-line tool `sha256`.

jlebar commented:
When you fix this, you do not need to create yet another PR.  Just make a new commit and force-push to this PR.

jlebar commented on WORKSPACE:
```diff
@@ -11,9 +11,9 @@ http_archive(
-    strip_prefix = "googletest-release-1.8.0",
+    url = "https://github.com/google/googletest/archive/master.zip",
+    sha256 = "b1c3ddb37427f9afcc8c91aee7b21826bf0cda54dc17d64c001a67ad2cf97250",
+    strip_prefix = "googletest-master",
```
I *think* this should also be `strip_prefix = "googletest-b1c3ddb..."` but I'm not sure.

BowNV committed:
improve WORKSPACE by using a certain commit ID

By running the command: "bazel run  //:cudnn_test --action_env=CUDNN_PATH=${VULCAN_INSTALL_DIR}/cuda --action_env=CUDA_PATH=/usr/local/cuda"

Building and  testing can be passed successfully as following
"
checking for a BSD-compatible install... /usr/bin/install -c
checking whether build environment is sane... yes
checking for a thread-safe mkdir -p... /bin/mkdir -p
checking for gawk... no
checking for mawk... mawk...
...
INFO: Build completed successfully, 327 total actions
INFO: Build completed successfully, 327 total actions
......
I0523 11:58:27.072669 12884 cudnn_util.cc:68] Running ...... on TITAN V
[==========] Running 7875 tests from 7 test suites.
[----------] Global test environment set-up.
[----------] 3 tests from ConvolutionTest
[ RUN      ] ConvolutionTest.GetWorkspaceSize_Overflow
[       OK ] ConvolutionTest.GetWorkspaceSize_Overflow (2 ms)
[ RUN      ] ConvolutionTest.GetConvolutionDesciptor_ArrayLengthRequested_Range
[       OK ] ConvolutionTest.GetConvolutionDesciptor_ArrayLengthRequested_Range (0 ms)
[ RUN      ] ConvolutionTest.GetGroupedConvolutionForwardOutputDim
[       OK ] ConvolutionTest.GetGroupedConvolutionForwardOutputDim (0 ms)
[----------] 3 tests from ConvolutionTest (3 ms total)
......

"

Thanks

BowNV commented:
Can you help to merge into master branch?

Thanks
Bo

chsigg commented:
Thanks BowNV.

I'm working on merging this PR, expect it to land in the next few days.

BowNV commented:
OK, THANKS A LOT!

chsigg commented:
This PR has been merged. Closing. Thanks for the fix!
